porcupine/plugins/anchors.py

```diff
@@ -0,0 +1,173 @@
+from __future__ import annotations
+import tkinter
+import time
+from typing import List
+from porcupine import tabs, get_tab_manager
+from porcupine.plugins.linenumbers import LineNumbers
+
+# Dependent on code from linenumbers.py
+setup_after = ["linenumbers"]
+
+
+class AnchorManager:
+    def __init__(self, tab_textwidget: tkinter.Text, linenumbers: tkinter.Canvas) -> None:
+        self.tab_textwidget = tab_textwidget
+        self.linenumbers = linenumbers
+
+        self.anchor_symbol = "¶"
+
+        # tkinter have default marks such as "insert", "current", "tkinter::anchor1"
+        self.custom_anchor_prefix = "anchor_"
+
+        linenumbers.bind("<<Updated>>", self.update_linenumbers, add=True)
+
+    def _get_anchors(self) -> List:
+        return [
+            anchor
+            for anchor in self.tab_textwidget.mark_names()
+            if anchor.startswith(self.custom_anchor_prefix)
+        ]
+
+    def _get_cursor_index(self):
+        return self.tab_textwidget.index("insert linestart")
+
+    def toggle_on_off(self, event: tkinter.Event[tkinter.Misc]) -> None:
+
+        self.prevent_duplicate_anchors()
+
+        cursor_index = self._get_cursor_index()
+        for anchor in self.tab_textwidget.mark_names():
+            if (
+                anchor.startswith(self.custom_anchor_prefix)
+                and self.tab_textwidget.index(anchor) == cursor_index
+            ):
```

Comment:
Nit: You can simplify this slightly by using `_get_anchors()`.

```suggestion
        for anchor in self._get_anchors():
            if self.tab_textwidget.index(anchor) == cursor_index:
```